lib/ansible/plugins/connection/psrp.py

```diff
@@ -580,6 +507,8 @@ def put_file(self, in_path: str, out_path: str) -> None:
             raise AnsibleFileNotFound('file or module does not exist: "%s"' % to_native(in_path))
 
         def read_gen():
+            yield from list(in_data.decode().splitlines())
```

Comment:
Isn't the `list()` redundant with the `yield from`? `splitlines()` should already be giving you an iterable that it's happy with...

Reply:
This was actually a recommendation from pylint but I think you are right, I'll try it out without the `list`.